packages/earn-controller/src/EarnController.ts

```diff
@@ -436,54 +576,80 @@ export class EarnController extends BaseController<
    * @returns A promise that resolves when the vault metadata has been updated
    */
-  async refreshPooledStakingVaultMetadata(): Promise<void> {
-    const chainId = this.#getCurrentChainId();
-    const vaultMetadata = await this.#stakingApiService.getVaultData(chainId);
+  async refreshPooledStakingVaultMetadata(chainId?: number): Promise<void> {
+    const chainIdToUse = chainId ?? this.#getCurrentChainId();
+    const vaultMetadata =
+      await this.#earnApiService.pooledStaking.getVaultData(chainIdToUse);
 
     this.update((state) => {
-      state.pooled_staking.vaultMetadata = vaultMetadata;
+      const chainState =
+        state.pooled_staking[chainIdToUse] ??
+        DEFAULT_POOLED_STAKING_CHAIN_STATE;
+      state.pooled_staking[chainIdToUse] = {
+        ...chainState,
+        vaultMetadata,
+      };
     });
   }
 
   /**
    * Refreshes pooled staking vault daily apys for the current chain.
    * Updates the pooled staking vault daily apys controller state.
    *
+   * @param chainId - The chain id to refresh pooled staking vault daily apys for (optional).
    * @param days - The number of days to fetch pooled staking vault daily apys for (defaults to 365).
    * @param order - The order in which to fetch pooled staking vault daily apys. Descending order fetches the latest N days (latest working backwards). Ascending order fetches the oldest N days (oldest working forwards) (defaults to 'desc').
    * @returns A promise that resolves when the pooled staking vault daily apys have been updated.
    */
   async refreshPooledStakingVaultDailyApys(
+    chainId?: number,
     days = 365,
     order: 'asc' | 'desc' = 'desc',
   ): Promise<void> {
-    const chainId = this.#getCurrentChainId();
-    const vaultDailyApys = await this.#stakingApiService.getVaultDailyApys(
-      chainId,
-      days,
-      order,
-    );
+    const chainIdToUse = chainId ?? this.#getCurrentChainId();
+    const vaultDailyApys =
+      await this.#earnApiService.pooledStaking.getVaultDailyApys(
+        chainIdToUse,
+        days,
+        order,
+      );
 
     this.update((state) => {
-      state.pooled_staking.vaultDailyApys = vaultDailyApys;
+      const chainState =
+        state.pooled_staking[chainIdToUse] ??
+        DEFAULT_POOLED_STAKING_CHAIN_STATE;
+      state.pooled_staking[chainIdToUse] = {
+        ...chainState,
+        vaultDailyApys,
+      };
     });
   }
 
   /**
    * Refreshes pooled staking vault apy averages for the current chain.
    * Updates the pooled staking vault apy averages controller state.
    *
+   * @param chainId - The chain id to refresh pooled staking vault apy averages for (optional).
    * @returns A promise that resolves when the pooled staking vault apy averages have been updated.
    */
-  async refreshPooledStakingVaultApyAverages() {
-    const chainId = this.#getCurrentChainId();
+  async refreshPooledStakingVaultApyAverages(chainId?: number) {
```

Comment:
What do you think about noting that `refreshPooledStakingVaultApyAverages` now takes an optional `chainId` argument in the changelog?